### PR TITLE
Add start and implement exercise sessions and save exercise sessions

### PR DESCRIPTION
**Pull request by ChubbsBunns**
Allows the user to start exercises and complete/ not complete them.
Completed exercises are saved as a file that will remain saved during the next run of Fitness Duke,

### CODE REVIEW

File: src/main/java/seedu/duke/storage/UserCareerData.java [hunk 1]
```diff
@@ -38,5 +40,19 @@ public ArrayList<Session> getTotalUserCareerSessions () {
...
+            System.out.println("Session " + (i+1));
+            String dateTime = totalUserCareerSessions.get(i).getDateAdded().toString();
+            String[] dateSplit = dateTime.split("T", 2);
+            assert dateSplit.length == 2;
+            System.out.println("On this date: " + dateSplit[0]);
+            exercisePrinter.printExercise(totalUserCareerSessions.get(i).getSessionExercises());
+            if (i != totalUserCareerSessions.size() - 1){
+                System.out.println("\n ");
+            }
+        }
```
Comment (EangJS) [inline src/main/java/seedu/duke/storage/UserCareerData.java:55]: Would it be better for this to be under Ui ?

File: src/main/java/seedu/duke/commands/CommandHandler.java [hunk 2]
```diff
@@ -3,45 +3,114 @@
...
+                            " Please generate a workout and use the \"start\" command!");
+                    break;
+                case "history":
+                    userCareerData.printAllFinishedWorkoutSessions();
+                    break;
+                default:
+                    ui.unknownCommand();
+                    errorExists = true;
+                    break;
+                }
```
Comment (EangJS) [inline src/main/java/seedu/duke/commands/CommandHandler.java:111]: Would it be better to abstract this ? Method is very long.

File: src/main/java/seedu/duke/commands/CommandHandler.java [hunk 3]
```diff
@@ -3,45 +3,114 @@
...
+import java.util.Scanner;
+
 public class CommandHandler {
+    private static final boolean COMPLETED_EXERCISE = true;
+    private static final boolean INCOMPLETE_EXERCISE = false;
 
-    public void handleUserCommands (String rawUserCommands, Ui ui, GenerateExercise exerciseGenerator,
-                                    UserCareerData userCareerData) {
+    /**
+     * @param rawUserCommands      This refers to the
```
Comment (L-K-Chng) [inline src/main/java/seedu/duke/commands/CommandHandler.java:16]: Good addition of javadoc comments

File: src/main/java/seedu/duke/commands/CommandHandler.java [hunk 4]
```diff
@@ -3,45 +3,114 @@
...
+                        ui.byeUser();
+                        System.exit(0);
+                    } else {
+                        System.out.println("You got this! Finish your exercise session!");
+                    }
+                    break;
+                case "readSample":
+                case "writeSample":
+                    throw new DukeError("Finish your exercise! Testing of our features can come after that :)");
+                case "start":
```
Comment (L-K-Chng) [inline src/main/java/seedu/duke/commands/CommandHandler.java:51]: Great addition of start, current, finish, cancel commands

**Review by L-K-Chng — APPROVED** [review]

**Pull request merged**